formidable/forms/field_builder.py

```diff
@@ -23,10 +23,17 @@ def __init__(self, field):
         self.validator_factory = self.validator_factory_class()
 
     def build(self, role=None):
-        self.access = self.field.accesses.all()[0] if role else None
+        self.access = self.get_accesses(role)
         field_class = self.get_field_class()
         return field_class(**self.get_field_kwargs())
 
+    def get_accesses(self, role):
+        if role:
+            # The role is previously "prefetch" in order to avoid database
+            # hit, we don't use a get() method in queryset.
+            return self.field.accesses.all()[0]
```

Comment:
what happens here if the field.accesses.all() is empty?

I'm not sure, but if ``field.accesses.all()`` is a queryset, you can use ``.first()`` that'll eventually return ``None`` if the queryset is empty. (but I don't know if it has side effects)